### ISSUE TITLE
Stream does not close when pulling an image

### ISSUE THREAD

**Issue by exdx** [vv9k/docker-api-rs#19]
- Crate version: 0.9.1
- OS: MacOS 12.3.1
- Output of running `docker version` on the command line:
```
Client: Docker Engine - Community
 Cloud integration: 1.0.12
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:13:00 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:15:47 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

When pulling an image, per the example in the repository, the image is able to be pulled successfully per the logs, but then the program just hangs, without moving out of the loop, at the end. Is there a way to close the stream or send None when the pull has finished? 

```
    let pull_opts = PullOpts::builder().image(config.image.clone()).build();
    let images = docker.images();
    let mut stream = images.pull(&pull_opts);

    while let Some(pull_result) = stream.next().await {
        match pull_result {
            Ok(output) => {
                println!("{:?}", output);
            },
            Err(e) => eprintln!("{}", e),
        }
    }
```

**Comment by vv9k** [comment]
The issue should be resolved as of https://github.com/vv9k/containers-api-conn/commit/b172a97ca7163c6236ec54d097e84ac59009aef6 That I just commited. The fix was in a separate crate so you might need to `cargo update` before running your project again to see the results.

I've tested using the example of this crate:
```shell
$ cargo r --example image pull centos:latest
```

Before this commit it would hang at the end and after it correctly ends the process.
